clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -1673,7 +1677,7 @@ mlir::Location CIRGenModule::getLocForFunction(const clang::FunctionDecl *FD) {
   bool invalidLoc = (FD->getSourceRange().getBegin().isInvalid() ||
                      FD->getSourceRange().getEnd().isInvalid());
   if (!invalidLoc)
-    return getLoc(FD->getSourceRange());
```

Comment:
I'm not sure if there is a specific need to a range loc for a function. Both MLIR and LLVM IR expect a single starting line for it.

Reply:
For static analysis / code mod it's important we know the range for the sake of giving scope based diagnostics and for applying whole function substitutions. My suggestion is that you leave this as is and extract the start `mlir::Location` when lowering to MLIR/LLVM.

Reply:
> For static analysis / code mod it's important we know the range for the sake of giving scope based diagnostics and for applying whole function substitutions. My suggestion is that you leave this as is and extract the start `mlir::Location` when lowering to MLIR/LLVM.

We can also fix the MLIR library to support the fused location case, but I'm not sure this is general enough to fit there?

Reply:
Switched to updating the loc of `LLVMFuncOp` instead of the CIR FuncOp.